daft/io/writer.py

```diff
@@ -175,8 +175,9 @@ def __init__(
         self.is_closed = False
 
     def _create_writer(self, schema: pa.Schema) -> pacsv.CSVWriter:
+        output_file = self.fs.open_output_stream(self.full_path)
         return pacsv.CSVWriter(
-            self.full_path,
+            output_file,
```

Comment:
The pacsv writer doesn't have a `filesystem` argument for us to pass in a fs to, instead we have to use the `open_output_stream` function on the fs, and pass it into the writer.